// @inkeep/ai-api/models/components/createchatsessionwithchatresultinput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateChatSessionWithChatResultInput$ = exports.ChatMode$ = exports.ChatModeOptions$ = exports.ChatModeOptions = void 0;
const chatsessioninput_1 = require("./chatsessioninput");
const z = __importStar(require("zod"));
var ChatModeOptions;
(function (ChatModeOptions) {
    ChatModeOptions["Turbo"] = "turbo";
    ChatModeOptions["Auto"] = "auto";
})(ChatModeOptions || (exports.ChatModeOptions = ChatModeOptions = {}));
/** @internal */
exports.ChatModeOptions$ = z.nativeEnum(ChatModeOptions);
/** @internal */
var ChatMode$;
(function (ChatMode$) {
    ChatMode$.inboundSchema = z.union([
        exports.ChatModeOptions$,
        z.string(),
    ]);
    ChatMode$.outboundSchema = z.union([
        exports.ChatModeOptions$,
        z.string(),
    ]);
})(ChatMode$ || (exports.ChatMode$ = ChatMode$ = {}));
/** @internal */
var CreateChatSessionWithChatResultInput$;
(function (CreateChatSessionWithChatResultInput$) {
    CreateChatSessionWithChatResultInput$.inboundSchema = z
        .object({
        integration_id: z.string(),
        chat_session: chatsessioninput_1.ChatSessionInput$.inboundSchema,
        chat_mode: z.union([exports.ChatModeOptions$, z.string()]).optional(),
        stream: z.boolean().default(false),
    })
        .transform((v) => {
        return {
            integrationId: v.integration_id,
            chatSession: v.chat_session,
            ...(v.chat_mode === undefined ? null : { chatMode: v.chat_mode }),
            stream: v.stream,
        };
    });
    CreateChatSessionWithChatResultInput$.outboundSchema = z
        .object({
        integrationId: z.string(),
        chatSession: chatsessioninput_1.ChatSessionInput$.outboundSchema,
        chatMode: z.union([exports.ChatModeOptions$, z.string()]).optional(),
        stream: z.boolean().default(false),
    })
        .transform((v) => {
        return {
            integration_id: v.integrationId,
            chat_session: v.chatSession,
            ...(v.chatMode === undefined ? null : { chat_mode: v.chatMode }),
            stream: v.stream,
        };
    });
})(CreateChatSessionWithChatResultInput$ || (exports.CreateChatSessionWithChatResultInput$ = CreateChatSessionWithChatResultInput$ = {}));
