// @inkeep/ai-api/models/components/recordtype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordType$ = exports.RecordTypeEnumerated$ = exports.RecordTypeEnumerated = void 0;
const z = __importStar(require("zod"));
var RecordTypeEnumerated;
(function (RecordTypeEnumerated) {
    RecordTypeEnumerated["Documentation"] = "documentation";
    RecordTypeEnumerated["Site"] = "site";
    RecordTypeEnumerated["DiscoursePost"] = "discourse_post";
    RecordTypeEnumerated["GithubIssue"] = "github_issue";
    RecordTypeEnumerated["GithubDiscussion"] = "github_discussion";
    RecordTypeEnumerated["StackoverflowQuestion"] = "stackoverflow_question";
    RecordTypeEnumerated["DiscordForumPost"] = "discord_forum_post";
    RecordTypeEnumerated["DiscordMessage"] = "discord_message";
    RecordTypeEnumerated["CustomQuestionAnswer"] = "custom_question_answer";
})(RecordTypeEnumerated || (exports.RecordTypeEnumerated = RecordTypeEnumerated = {}));
/** @internal */
exports.RecordTypeEnumerated$ = z.nativeEnum(RecordTypeEnumerated);
/** @internal */
var RecordType$;
(function (RecordType$) {
    RecordType$.inboundSchema = z.union([
        exports.RecordTypeEnumerated$,
        z.string(),
    ]);
    RecordType$.outboundSchema = z.union([
        exports.RecordTypeEnumerated$,
        z.string(),
    ]);
})(RecordType$ || (exports.RecordType$ = RecordType$ = {}));
