// @inkeep/ai-api/models/components/record.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordT$ = void 0;
const recordtype_1 = require("./recordtype");
const z = __importStar(require("zod"));
/** @internal */
var RecordT$;
(function (RecordT$) {
    RecordT$.inboundSchema = z
        .object({
        type: recordtype_1.RecordType$.inboundSchema,
        url: z.nullable(z.string()).optional(),
        title: z.nullable(z.string()).optional(),
        description: z.nullable(z.string()).optional(),
        breadcrumbs: z.nullable(z.array(z.string())).optional(),
    })
        .transform((v) => {
        return {
            type: v.type,
            ...(v.url === undefined ? null : { url: v.url }),
            ...(v.title === undefined ? null : { title: v.title }),
            ...(v.description === undefined ? null : { description: v.description }),
            ...(v.breadcrumbs === undefined ? null : { breadcrumbs: v.breadcrumbs }),
        };
    });
    RecordT$.outboundSchema = z
        .object({
        type: recordtype_1.RecordType$.outboundSchema,
        url: z.nullable(z.string()).optional(),
        title: z.nullable(z.string()).optional(),
        description: z.nullable(z.string()).optional(),
        breadcrumbs: z.nullable(z.array(z.string())).optional(),
    })
        .transform((v) => {
        return {
            type: v.type,
            ...(v.url === undefined ? null : { url: v.url }),
            ...(v.title === undefined ? null : { title: v.title }),
            ...(v.description === undefined ? null : { description: v.description }),
            ...(v.breadcrumbs === undefined ? null : { breadcrumbs: v.breadcrumbs }),
        };
    });
})(RecordT$ || (exports.RecordT$ = RecordT$ = {}));
